src/fu-util.c

```diff
@@ -1043,6 +1064,26 @@ fu_util_device_test_filename(FuUtilPrivate *priv,
 		json_builder_set_member_name(helper->builder, "interactive");
 		json_builder_add_boolean_value(helper->builder, interactive);
 	}
+	if (json_object_has_member(json_obj, "cpu-architectures")) {
+		JsonArray *json_array = json_object_get_array_member(json_obj, "cpu-architectures");
+		gboolean matched = FALSE;
+		const gchar *arch = g_hash_table_lookup(helper->report_metadata, "CpuArchitecture");
+		for (guint i = 0; i < json_array_get_length(json_array); i++) {
+			const gchar *arch_tmp = json_array_get_string_element(json_array, i);
+			if (g_strcmp0(arch, arch_tmp) == 0) {
+				matched = TRUE;
+				break;
+			}
+		}
+		if (!matched) {
+			fu_console_print_full(priv->console,
+					      FU_CONSOLE_PRINT_FLAG_WARNING,
+					      "%s [%s]\n",
+					      /* TRANSLATORS: tests can be specific to a CPU */
+					      _("Skipping due to incorrect architecture", arch);
```

Comment:
I think you're missing a `)` somewhere in here

Reply:
I've pushed something better; a skipped test is not the same as a successful or failed test, so I've created a new counter.